doc/services/pm/device_runtime.rst

```diff
@@ -126,7 +126,7 @@ by the PM subsystem to suspend or resume devices.
     static int mydev_pm_action(const struct device *dev,
                                enum pm_device_action *action)
     {
-        switch (action) {
+        switch (*action) {
```

Comment:
This is the wrong fix. You have to change the function signature. See the callback type:
https://github.com/zephyrproject-rtos/zephyr/blob/0c7e87714d923d5e1db89e0fb15fbdf45dcd4a00/include/zephyr/pm/device.h#L119

Reply:
Well spotted, thank you.